### DISCUSSION TITLE
Body click events on windows notification is lost when using TaskbarIcon

### DISCUSSION THREAD

**Discussion by sonnin** [HavenDV/H.NotifyIcon#136] [Q&A] [answered]
Hi,

My WinUI applications is using Microsoft.Toolkit.Uwp.Notifications and Windows.UI.Notifications to create notifications in windows action center. And when clicking on the windows notification body/text area, I get an event on the Windows.UI.Notifications.ToastNotification.Dismissed EventHandler, with a UserCancelled reason.
Now after including the H.NotifyIcon.WinUI dependency and using the TaskbarIcon functionality, the application no longer gets an event when clicking on the body/text of the notification. When I clear the notification with the X I do get an event. Further when when clicking on the notification in windows action center, then notification disappears at first, but closing and opening the notification center again, the notification reappears.

My question are therefore, how does this library override my event handler and how can I tell the library not to do this.

Where in the H.NotifyIcon source code are notification callback/events handles registered ?

thanks
Sonni

**Comment by HavenDV** [comment]
To receive system messages, the library registers a hidden window. I think this window may somehow influence other code, but in this particular case nothing comes to mind. This simply creates events on the NIN_BALLOONSHOW/NIN_BALLOOONHIDE messages without catching them inclusively in any way.
Perhaps some PInvoke.CreateWindowEx settings can help avoid this, but I'm not sure

https://github.com/HavenDV/H.NotifyIcon/blob/master/src/libs/H.NotifyIcon/Core/MessageWindow.cs

**Reply by sonnin** [reply]
When I debug the example WinUI application I can see that MessageWindow.ProcessWindowMessage is called with PInvoke.NIN_BALLOONUSERCLICK when I click on the notification in windows action center. But the information about available is mouse click cordinates.

**Comment by HavenDV** [comment]
The TrayIcon class has ShowNotification/ClearNotifications methods, which, like TrayIcon, work through the system API:
https://learn.microsoft.com/en-us/windows/win32/api/shellapi/nf-shellapi-shell_notifyicona
Maybe this can help somehow

**Comment by sonnin** [comment, marked as answer]
Hi, 
I turned out to be because I changed the <AssemblyName> in the project file. And at the same time <com:ExeServer  Executable ..> was still referring to the old exe filename. 
Once these two are the same toast callback worked again.
